src/extension.ts

```diff
@@ -1,398 +1,194 @@
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterProposeds`, async () => await filterState(logs, context, astResultsProvider, StateLevel.proposed, PROPOSED_FILTER)));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterConfirmeds`, async () => await filterState(logs, context, astResultsProvider, StateLevel.confirmed, CONFIRMED_FILTER)));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterToVerifies`, async () => await filterState(logs, context, astResultsProvider, StateLevel.toVerify, TO_VERIFY_FILTER)));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterUrgents`, async () => await filterState(logs, context, astResultsProvider, StateLevel.urgent, URGENT_FILTER)));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterIgnoreds`, async () => await filterState(logs, context, astResultsProvider, StateLevel.ignored, IGNORED_FILTER)));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.filterNotIgnoreds`, async () => await filterState(logs, context, astResultsProvider, StateLevel.notIgnored, NOT_IGNORED_FILTER)));
-
-    // Pickers command
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.generalPick`, async () => {
-        await multiStepInput(logs, context);
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.projectPick`, async () => {
-        await projectPicker(context, logs);
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.branchPick`, async () => {
-        await branchPicker(context, logs);
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.scanPick`, async () => {
-        await scanPicker(context, logs);
-    }));
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.scanInput`, async () => {
-        await scanInput(context, logs);
-    }));
-
-    // Visual feedback on wrapper errors
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.showError`, () => {
-        vscode.window.showErrorMessage(getError(context)!);
-    }));
-
-    // Kics remediation command
-    context.subscriptions.push(vscode.commands.registerCommand(`${EXTENSION_NAME}.kicsRemediation`, async (fixedResults, kicsResults, file, diagnosticCollection, fixAll, fixLine) => {
-        await kicsProvider.kicsRemediation(fixedResults, kicsResults, file, diagnosticCollection, fixAll, fixLine, logs);
-    }));
-    scaResultsProvider.refreshData(SCA_START_SCAN);
+  });
+  // Register Settings
+  const commonCommand = new CommonCommand(context, logs);
+  commonCommand.registerSettings();
+  kicsScanCommand.registerSettings();
```

Comment:
i think this should be before the commandCommand actions.

Reply:
just grouped the register settings together